client/pages/DataExport/ResultsTable/ResultsTable.tsx

```diff
@@ -29,21 +30,22 @@ const ResultsTable: React.FC<{ tableName: string }> = ({ tableName }) => {
     cycleName: string
     section: string
   }>()
+
+  const cycle = useCycle()
   const selection = useDataExportSelection(assessmentSection)
   const countries = useDataExportCountries()
   const tableSections = useTableSections({ sectionName: assessmentSection })
 
   const { tables } = tableSections.find((tableSection) => tableSection.tables.find((table) => table.props.dataExport))
   const table = tables.find((table) => table.props.dataExport)
   const baseUnit = table?.props?.unit
-  const columns = selection.sections[assessmentSection].columns ?? []
+  const columns = table?.props?.columnsExport[cycle.uuid] ?? []
```

Comment:
Are you sure you're exporting the Columns selected by the user ?

Reply:
or shouldn't this be ```table.props.columnsExport[cycle.uuid] ?? table.props.columnNames[cycle.uuid]```?
or shouldn't the columns be the same that the ones initiated in DataExport?